File: source/go/internal/jyotish/prashna.go
package jyotish

import (
	"github.com/ishankgupta95/panchang/source/go/v5/internal/astronomy"
	"github.com/ishankgupta95/panchang/source/go/v5/internal/types"
)

func ComputePrashnaChart(
	ctx *astronomy.EphemerisCtx,
	questionMomentMs int64,
	location types.GeoLocation,
	options BirthChartOptions,
) (types.BirthChart, error) {
	prashnaOptions := options
	if prashnaOptions.HouseSystem == "" {
		prashnaOptions.HouseSystem = types.HouseSystemPlacidusKP
	}
	if prashnaOptions.Ayanamsa == "" {
		prashnaOptions.Ayanamsa = types.Krishnamurti
	}
	return ComputeRashiChart(ctx, questionMomentMs, location, prashnaOptions)
}

File: source/ts/src/jyotish/prashna.ts
import { computeRashiChart } from './charts';
import type { BirthChartOptions } from '../types/options';
import type { GeoLocation } from '../types/location';
import type { BirthChart } from '../types/jyotish';

/** Prashna (horary) chart for the moment a question is asked, with KP defaults (`'placidus-kp'` houses, `'krishnamurti'` ayanamsa) that `options` can override. */
export function computePrashnaChart(
  questionMoment: Date,
  location: GeoLocation,
  options: BirthChartOptions = {},
): BirthChart {
  const prashnaOptions: BirthChartOptions = {
    ...options,
    houseSystem: options.houseSystem ?? 'placidus-kp',
    ayanamsa: options.ayanamsa ?? 'krishnamurti',
  };
  return computeRashiChart(questionMoment, location, prashnaOptions);
}
